stratosphere/ams_mitm/source/set_mitm/setsys_mitm_service.cpp

```diff
@@ -102,6 +184,44 @@ namespace ams::mitm::settings {
         return GetFirmwareVersionImpl(out.GetPointer(), m_client_info);
     }
 
+    Result SetSysMitmService::SetBluetoothDevicesSettings(const sf::InMapAliasArray<SetSysBluetoothDevicesSettings> &settings) {
+        /* Forward to session unless external database setting enabled. */
+        R_UNLESS(ExternalBluetoothDatabaseEnabled(), sm::mitm::ResultShouldForwardToSession());
+
+        /* Create the external database if it doesn't exist. */
+        if (!HasExternalBluetoothDatabase()) {
+            R_TRY(fs::CreateFile(g_external_bluetooth_db_location, 0));
+        }
+
+        /* Backup local database to sd card. */
+        R_TRY(WriteExternalBluetoothDatabase(settings.GetPointer(), settings.GetSize()));
+
+        /* Also allow the updated database to be stored to system save as usual. */
+        return sm::mitm::ResultShouldForwardToSession();
+    }
+
+    Result SetSysMitmService::GetBluetoothDevicesSettings(sf::Out<s32> out_count, const sf::OutMapAliasArray<SetSysBluetoothDevicesSettings> &out) {
+        /* Forward to session unless external database setting enabled. */
+        R_UNLESS(ExternalBluetoothDatabaseEnabled(), sm::mitm::ResultShouldForwardToSession());
+
+        if (!HasExternalBluetoothDatabase()) {
+            /* Forward to the real command to fetch database stored in system save. */
+            R_TRY(setsysGetBluetoothDevicesSettingsFwd(m_forward_service.get(), out_count.GetPointer(), out.GetPointer(), out.GetSize()));
+
+            /* Create the external database file. */
+            R_TRY(fs::CreateFile(g_external_bluetooth_db_location, 0));
+
+            /* Backup local database to sd card. */
+            R_TRY(WriteExternalBluetoothDatabase(out.GetPointer(), out_count.GetValue()));
+        }
+        else {
+            /* Read the external database from sd card. */
+            R_TRY(ReadExternalBluetoothDatabase(out.GetPointer(), out.GetSize(), reinterpret_cast<size_t*>(out_count.GetPointer())));
```

Comment:
This is an invalid reinterpret_cast and will corrupt memory by writing 8 bytes to a 32-bit storage.

Also, `reinterpret_cast<size_t *>` preferred to `reinterpret_cast<size_t*>`.

Make the function take in the correct pointer size, do not reinterpret_cast pointers of different sizes in this way.

Also, I don't understand why you switched around the parameter order, it should be s32 *, then buffer.